### PR TITLE
Discovery fixes (#432, #434, #462, #463)

### LINKED ISSUE #432: Rework Discover feature according to https://github.com/status-im/status-react/wiki/Discover

### LINKED ISSUE #434: Discover double shown messages, message ordering & cannot tap users
cannot tap any users in discover and start chat / send contact request

Message ordering is wrong in Recents
this should have been;
Jarrad: #buying #eth and #milk 
Roman: i'm #buying eth also
Jarrad: #buying #eth

screenshot_20161109-211510

double shown messages and blank space

screenshot_20161109-211235

### LINKED ISSUE #462: Cannot copy address or public key in profile
Addresses and public key strings should be copy-able.
Also "Show QR" does not use the show QR screen that we have?

### LINKED ISSUE #463: new line in status causes crash
I pressed "Enter" on keyboard while editing status in drawer, then swiped away.

This causes a crash on my device and on my contacts device.

### PR DESCRIPTION
**Pull request by alwx**
This PR also includes:
1) Layout fixes for discovery screen (looks much closer to Zeplin sketches than before);
2) Layout fixes for contact screen (correct shadow/line, correct fonts on iOS);
3) Blue tags on discovery screen;
4) :pending contacts work much better than before (no more changing of pending status without approval).

fix #432 
fix #434 
fix #462 
fix #463

### CODE REVIEW

**Comment by jarradh** [comment]
Please add a fix for #463 - to remove newlines from statuses

File: src/status_im/discovery/handlers.cljs [hunk 1]
```diff
@@ -5,111 +5,115 @@
...
+      (protocol/send-discoveries-response!
+        {:web3        web3
+         :discoveries (discoveries/get-all :asc)
+         :message     {:from current-public-key
+                       :to   to}}))))
+
+(register-handler :discoveries-request-received
+  (u/side-effect!
+    (fn [_ [_ {:keys [from]}]]
+      (dispatch [:discoveries-send-portions from]))))
```
Comment (rasom) [inline src/status_im/discovery/handlers.cljs]: Does it mean that anyone can request discoveries and receive this response?

**Comment by rasom** [comment]
Can not add a contact. I will check, maybe we have the same problem in `develop`

cntct

**Comment by rasom** [comment]
Currently, discoveries are requested only when we start application
https://github.com/status-im/status-react/blob/13b8f697335b96c0c9f1b6d8fd2382b62a6b7122/src/status_im/handlers.cljs#L67-L67

And new statuses will be sent to direct contacts immediately, as I understood from the code. 
But imagine that A has B in contacts, and B has C in contacts. If C updates status, A will receive this status from B only after restarting. 

@alwx Do I confuse something, or that's how it works now?

File: src/status_im/discovery/handlers.cljs (hunk 1 above)
Comment (alwx) [inline src/status_im/discovery/handlers.cljs]: I've updated this method and the way it works

**Comment by alwx** [comment]
@rasom yes, you *were* right. I've updated it and now it also requests discoveries periodically.

**Comment by rasom** [comment]
Seems that migrations doesn't work properly.
Actually we can ignore it this time but...

screen shot 2016-11-17 at 13 46 35

**Comment by alwx** [comment]
No, wait. Probably it is a rebase-related issue. Will fix in 5 mins

**Comment by rasom** [comment]
This happens when i'm adding a contact

ac

**Comment by rasom** [comment]
User can start chat with himself

cc

**Pull request merged**